src/main/java/uk/gov/pay/adminusers/service/GovUkPayAgreementService.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.pay.adminusers.service;
+
+import com.google.inject.Inject;
+import org.slf4j.Logger;
+import uk.gov.pay.adminusers.logger.PayLoggerFactory;
+import uk.gov.pay.adminusers.model.GovUkPayAgreement;
+import uk.gov.pay.adminusers.persistence.dao.GovUkPayAgreementDao;
+import uk.gov.pay.adminusers.persistence.entity.GovUkPayAgreementEntity;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+import static java.lang.String.format;
+
+public class GovUkPayAgreementService {
+
+    private static Logger logger = PayLoggerFactory.getLogger(GovUkPayAgreementService.class);
+    private final GovUkPayAgreementDao agreementDao;
+    
+    @Inject
+    public GovUkPayAgreementService(GovUkPayAgreementDao govUkPayAgreementDao) {
+        this.agreementDao = govUkPayAgreementDao;
+    }
+
+
+    public Optional<GovUkPayAgreement> findGovUkPayAgreementByServiceId(int serviceId) {
```

Comment:
Is this an `int` because it’s the service internal (database) ID? I think we want to use the external ID everywhere in the code (the internal ID should only be used within the database). Might need to tweak the DAO and entity to make that work. Have a look at `InviteDao.findAllByServiceId(String)` for inspiration.

Reply:
Again, fair dinkum. I'll have to change `StripeAgreementService` I assume.